clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/provider/KubernetesV2InstanceProvider.java

```diff
@@ -82,49 +85,76 @@ public KubernetesV2Instance getInstance(String account, String location, String
   }
 
   @Override
-  public List<ContainerLog> getConsoleOutput(String account, String location, String fullName) {
-    Optional<KubernetesV2Credentials> optionalCredentials = accountResolver.getCredentials(account);
+  public List<ContainerLog> getConsoleOutput(String account, String namespace, String fullName) {
+    final Optional<KubernetesV2Credentials> optionalCredentials =
+        accountResolver.getCredentials(account);
     if (!optionalCredentials.isPresent()) {
       log.warn("Failure getting account {}", account);
       return null;
     }
-    KubernetesV2Credentials credentials = optionalCredentials.get();
 
-    Pair<KubernetesKind, String> parsedName;
+    final KubernetesV2Credentials credentials = optionalCredentials.get();
+    final Pair<KubernetesKind, String> parsedName;
+
     try {
       parsedName = KubernetesManifest.fromFullResourceName(fullName);
     } catch (Exception e) {
       return null;
     }
 
-    String name = parsedName.getRight();
-
-    V1Pod pod =
+    final String podName = parsedName.getRight();
+    final V1Pod pod =
         KubernetesCacheDataConverter.getResource(
-            credentials.get(KubernetesKind.POD, location, name), V1Pod.class);
-
-    List<ContainerLog> result = new ArrayList<>();
+            credentials.get(KubernetesKind.POD, namespace, podName), V1Pod.class);
 
     // Short-circuit if pod cannot be found
     if (pod == null) {
-      result.add(
+      return Lists.newArrayList(
           new ContainerLog("Error", "Failed to retrieve pod data; pod may have been deleted."));
-      return result;
     }
 
-    // Make live calls rather than abuse the cache for storing all logs
-    for (V1Container container : pod.getSpec().getContainers()) {
-      ContainerLog log = new ContainerLog();
-      log.setName(container.getName());
-      try {
-        log.setOutput(credentials.logs(location, name, container.getName()));
-      } catch (KubectlJobExecutor.KubectlException e) {
-        // Typically happens if the container/pod isn't running yet
-        log.setOutput(e.getMessage());
-      }
-      result.add(log);
-    }
+    return getPodLogs(credentials, pod);
+  }
+
+  @Nonnull
+  private List<ContainerLog> getPodLogs(
+      @Nonnull final KubernetesV2Credentials credentials, @Nonnull final V1Pod pod) {
+    final List<V1Container> initContainers = pod.getSpec().getInitContainers();
+    final List<V1Container> containers = pod.getSpec().getContainers();
 
+    final List<ContainerLog> result = new ArrayList<>(containers.size() + initContainers.size());
```

Comment:
This might be a bit clearer using streams (which in a number of places in our code are overused, but which I think could help here).  Something like the following could eliminate a lot of the intermediate arrays, and also remove the need for the `getContainerLogs` which is just looping over containers:
```
return Stream.concat(initContainers.stream(), containers.stream())
    .map(container -> getContainerLog(credentials, pod, container))
    .collect(Collectors.toList());
```

Reply:
I'll update this in the morning with the above.

Reply:
I think this made it a lot more clear, thanks for the tip.